taverna-cwl-utilities/src/main/java/org/apache/taverna/cwl/utilities/CWLUtil.java

```diff
@@ -0,0 +1,216 @@
+package org.apache.taverna.cwl.utilities;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+
+public class CWLUtil {
+
+	private static final String INPUTS = "inputs";
+	private static final String OUTPUTS = "outputs";
+	private static final String ID = "id";
+	private static final String TYPE = "type";
+	private static final String ARRAY = "array";
+	private static final String DESCRIPTION = "description";
+	private static final int DEPTH_0 = 0;
+	private static final int DEPTH_1 = 1;
+	private static final int DEPTH_2 = 2;
+
+	private static final String FLOAT = "float";
+	private static final String NULL = "null";
+	private static final String BOOLEAN = "boolean";
+	private static final String INT = "int";
+	private static final String DOUBLE = "double";
+	private static final String STRING = "string";
+	private static final String LABEL = "label";
+	private static final String FILE = "file";
+	private static final String FORMAT = "format";
+	private JsonNode nameSpace;
+	private JsonNode cwlFile;
+
+	public CWLUtil(JsonNode cwlFile) {
+		this.cwlFile = cwlFile;
+		processNameSpace();
+	}
+
+	public JsonNode getNameSpace() {
+		return nameSpace;
+	}
+
+	public void processNameSpace() {
+
+		if (cwlFile.has("$namespaces")) {
+			nameSpace = cwlFile.path("$namespaces");
+		}
+
+	}
+
+	public HashMap<String, Integer> processInputDepths() {
+		return process(cwlFile.get(INPUTS));
+	}
+
+	public HashMap<String, Integer> processOutputDepths() {
+		return process(cwlFile.get(OUTPUTS));
+	}
+
+	public HashMap<String, PortDetail> processInputDetails() {
+		return processdetails(cwlFile.get(INPUTS));
+	}
+
+	public HashMap<String, PortDetail> processOutputDetails() {
+		return processdetails(cwlFile.get(OUTPUTS));
+	}
+
+	public HashMap<String, Integer> process(JsonNode inputs) {
```

Comment:
It is unclear what this public `process()` method does or returns... could you add a `/**` javadocs explaining inputs and outputs for these methods?

Also as mentioned before, `HashMap` -> `Map` :-)